src/prefect/flows.py

```diff
@@ -476,6 +482,169 @@ def serialize_parameters(self, parameters: Dict[str, Any]) -> Dict[str, Any]:
+        tags: Optional[List[str]] = None,
+        version: Optional[str] = None,
+        pause_on_shutdown: bool = True,
+        print_starting_message: bool = True,
+    ):
+        """
+        Creates a deployment for this flow and starts a runner to monitor for scheduled work.
+
+        Args:
+            name: The name to give the created deployment.
+            interval: An interval on which to execute the current flow. Accepts either a number
+                or a timedelta object. If a number is given, it will be interpreted as seconds.
+            cron: A cron schedule of when to execute runs of this flow.
+            rrule: An rrule schedule of when to execute runs of this flow.
+            triggers: A list of triggers that should kick of a run of this flow.
+            parameters: A dictionary of default parameter values to pass to runs of this flow.
+            description: A description for the created deployment. Defaults to the flow's
+                description if not provided.
+            tags: A list of tags to associate with the created deployment for organizational
+                purposes.
+            version: A version for the created deployment. Defaults to the flow's version.
+            pause_on_shutdown: If True, provided schedule will be paused when the serve function is stopped.
+                If False, the schedules will continue running.
+            print_starting_message: Whether or not to print the starting message when flow is served.
+
+        Examples:
+            Serve a flow:
+
+            ```python
+            from prefect import flow
+
+            @flow
+            def my_flow(name):
+                print(f"hello {name}")
+
+            if __name__ == "__main__":
+                my_flow.serve(__file__)
+            ```
+
+            Serve a flow and run it every hour:
+
+            ```python
+            from prefect import flow
+
+            @flow
+            def my_flow(name):
+                print(f"hello {name}")
+
+            if __name__ == "__main__":
+                my_flow.serve(__file__, interval=3600)
+            ```
+        """
+        from prefect.runner import Runner
+
+        # Handling for my_flow.serve(__file__)
+        # Will set name to name of file where my_flow.serve() without the extension
+        # Non filepath strings will pass through unchanged
+        name = Path(name).stem
+
+        runner = Runner(name=name, pause_on_shutdown=pause_on_shutdown)
+        deployment_id = await runner.add(
+            self,
+            name=name,
+            triggers=triggers,
+            interval=interval,
+            cron=cron,
+            rrule=rrule,
+            parameters=parameters,
+            description=description,
+            tags=tags,
+            version=version,
+        )
+        if print_starting_message:
+            help_message = (
+                f"[green]Your flow {self.name!r} is served and polling for scheduled"
```

Comment:
"Is served" scans a little odd to me -- maybe "is being served"? (Also in CLI help message$